src/atoms/TextInput/TextInput.web.js

```diff
@@ -1,3 +1,486 @@
+          return '240px';
+      }
+    },
+  },
+};
+
+const InputContainer = styled(View)`
+  background-color: transparent;
+`;
+
+const StyledInput = styled.input`
+  font-size: ${(props) => props.theme.fonts.size.medium};
+  line-height: ${(props) => props.theme.fonts.lineHeight.medium};
+  font-family: ${(props) => props.theme.fonts.family.lato.regular};
+  color: ${styles.textInput.color};
+  border: none;
+  background-color: transparent;
+  pointer-events: ${(props) => (props.disabled ? 'none' : '')};
+  &::selection {
+    background-color: ${(props) => props.onSelect};
+  }
+  &::-webkit-outer-spin-button {
+    appearance: none;
+    margin: 0;
+  }
+  &::-webkit-inner-spin-button {
+    appearance: none;
+    margin: 0;
+  }
+  &[type='number'] {
+    appearance: textfield;
+  }
+  &:hover {
+    background-color: ${styles.textInput.hoverBackgroundColor};
+  }
+  &:focus {
+    outline: none;
+  }
+  &:active {
+    outline: none;
+  }
+  &::placeholder {
+    color: ${(props) => props.placeholderTextColor};
+  }
+  /* Removes red box shadow rectangle on firefox */
+  &:invalid {
+    box-shadow: none;
+  }
+`;
+
+const FillContainer = styled(View)`
+  background-color: ${styles.fillContainer.backgroundColor};
+  border-top-left-radius: 2px;
+  border-top-right-radius: 2px;
+  margin-top: auto;
+  position: relative;
+  box-sizing: border-box;
+  &:hover {
+    background-color: ${styles.fillContainer.hoverBackgroundColor};
+  }
+`;
+
+const getAccessoryConfig = ({ errorText, prefix, suffix, iconLeft, iconRight }) => {
+  const hasError = !isEmpty(errorText);
+  const hasPrefix = !isEmpty(prefix);
+  const hasSuffix = !isEmpty(suffix);
+  const hasLeftIcon = !hasPrefix && !isEmpty(iconLeft);
+  const hasRightIcon = !hasSuffix && !isEmpty(iconRight);
+  return { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon };
+};
+
+const getPlaceholderTextColor = ({ theme, disabled, isPlaceholderVisible, hasAnimatedLabel }) => {
+  if (isPlaceholderVisible || !hasAnimatedLabel) {
+    if (disabled) {
+      return theme.colors.shade[930];
+    } else {
+      return theme.colors.shade[940];
+    }
+  } else {
+    return 'transparent';
+  }
+};
+
+// eslint-disable-next-line complexity
+const TextInput = ({
+  placeholder,
+  onChange,
+  onBlur,
+  helpText,
+  successText,
+  errorText,
+  variant,
+  prefix,
+  suffix,
+  disabled,
+  value,
+  iconLeft,
+  iconRight,
+  maxLength,
+  label,
+  testID,
+  labelPosition,
+  width,
+  type,
+  id,
+  name,
+}) => {
+  const theme = useContext(ThemeContext);
+  const inputRef = useRef();
+  const containerRef = useRef();
+  const [isFocused, setIsFocused] = useState(false);
+  const [input, setInput] = useState(value || '');
+  // Used for storing layout value of TextInput
+  const [layoutDimensions, setLayoutDimensions] = useState(null);
+  // Used to hide placeholder while label is inside the TextInput
+  const [isPlaceholderVisible, setIsPlaceholderVisible] = useState(isFocused);
```

Comment:
Initial value for this will always stay `false`, right? Because initial value of `isFocused` is `false`. 🤔